### PR TITLE
adds initial unit testing

### PR DESCRIPTION
**Pull request by Inveracity**
```
pip install requirements-ci.txt
pytest app_test.py --cov .
```
output
```
===== test session starts =====

platform linux -- Python 3.6.3, pytest-3.4.0, py-1.5.2, pluggy-0.6.0
rootdir: /vagrant, inifile:
plugins: cov-2.5.1
collected 2 items

app_test.py .. [100%]

----------- coverage: platform linux, python 3.6.3-final-0 -----------
Name                                           Stmts   Miss  Cover
------------------------------------------------------------------
app.py                                             5      3    40%
app_test.py                                       16      0   100%
deploy.py                                         13     13     0%
gunicorn_config.py                                30     30     0%
pysite/__init__.py                                 7      1    86%
pysite/base_route.py                              59     24    59%
pysite/constants.py                               14      0   100%
pysite/database.py                                92     50    46%
pysite/decorators.py                              44     30    32%
pysite/mixins.py                                  21      4    81%
pysite/route_manager.py                           52     22    58%
pysite/views/__init__.py                           0      0   100%
pysite/views/api/__init__.py                       0      0   100%
pysite/views/api/bot/__init__.py                   0      0   100%
pysite/views/api/bot/tag.py                       25     13    48%
pysite/views/api/bot/user.py                      17      2    88%
pysite/views/api/healthcheck.py                    7      1    86%
pysite/views/api/index.py                          7      1    86%
pysite/views/main/__init__.py                      0      0   100%
pysite/views/main/datadog.py                       7      1    86%
pysite/views/main/error_handlers/__init__.py       0      0   100%
pysite/views/main/error_handlers/http_404.py       7      0   100%
pysite/views/main/error_handlers/http_5xx.py       7      1    86%
pysite/views/main/index.py                         6      0   100%
pysite/views/main/invite.py                        7      1    86%
pysite/views/main/ws_test.py                       7      1    86%
pysite/views/staff/__init__.py                     0      0   100%
pysite/views/staff/index.py                        6      1    83%
pysite/views/ws/__init__.py                        0      0   100%
pysite/views/ws/echo.py                           17      7    59%
pysite/websockets.py                              29     13    55%
------------------------------------------------------------------
TOTAL                                            502    219    56%
===== 2 passed in 0.44 seconds =====
```

### CODE REVIEW

File: app_test.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from app import app
+
+from flask_testing import TestCase
+
+
+class siteTest(TestCase):
```
Comment (lemonsaurus) [inline app_test.py]: should be PascalCase

File: app_test.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
```
Comment (lemonsaurus) [inline app_test.py]: I'd like to see class docstrings.

File: app_test.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    def create_app(self):
+        app.config['TESTING'] = True
+        app.config['LIVESERVER_TIMEOUT'] = 10
+        app.allow_subdomain_redirects = True
+        return app
+
+
+class RootEndpoint(siteTest):
```
Comment (lemonsaurus) [inline app_test.py]: While PEP8 doesn't really specify whether or not to put a blank line after the class definition line, we must at least be consistent about whether we do or not.

File: app_test.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    def test_index(self):
+        response = self.client.get('/')
+        self.assertEqual(response.status_code, 200)
+
+    def test_api(self):
+        response = self.client.get('/nonexistantpath')
```
Comment (lemonsaurus) [inline app_test.py]: _existent_

File: app_test.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        self.assertEqual(response.status_code, 404)
```
Comment (lemonsaurus) [inline app_test.py]: gross, we've inherited camelCased methods.

**Review by lemonsaurus — CHANGES_REQUESTED** [review]
This looks pretty straight forward to use. I look forward to it. Thanks for writing it. :+1: :100: :1st_place_medal:

**Review by gdude2002 — APPROVED** [review]
👍

**Review by lemonsaurus — APPROVED** [review]

**Pull request merged**